### ISSUE TITLE
Deadlock in usage of DistributedQueryRunner

### ISSUE THREAD

**Issue by aweisberg** [prestodb/presto#15921]
@pettyjamesm has done some root causing, but hasn't been able to chase it down completely.

**Comment by aweisberg** [comment]
>  The problem is that because the read lock is not an “ownable” lock in JVM parlance, there’s no way to track which threads are holding those, but we know that any other test method that is executing a sql statement will be holding it
> 
> ```
>         com.facebook.presto.tests.TestDistributedSpilledQueriesWithTempStorage::testAccessControl running for 8.07m
> 	com.facebook.presto.tests.TestSpilledAggregationsWithTemporaryStorage running for 11.81m
> 	com.facebook.presto.tests.TestDistributedSpilledQueriesWithTempStorage running for 8.21m
> 	com.facebook.presto.tests.TestDistributedQueriesWithDynamicFilter running for 8.40m
> 	com.facebook.presto.tests.TestDistributedSpilledQueries running for 8.38m
> 	com.facebook.presto.tests.TestDistributedSpilledQueriesWithTempStorage::emptyJoins running for 8.09m.
> ```
> Access control + emptyJoins
> Suggestive of something being “stuck” in the presto server side and not the test threads themselves

**Comment by rschlussel** [comment]
I think it's something about the joins. Previous similar issues showed up in spill tests with joins https://github.com/prestodb/presto/issues/15542.  I haven't figured out the root cause and haven't even been able to reproduce locally. In the meantime I've disabled the emptyJoins() test here: https://github.com/prestodb/presto/pull/15898

**Comment by pettyjamesm** [comment]
I got some more time to dig into this, it seems like `TestDistributedSpilledQueries::emptyJoins()` and specifically [this query](https://github.com/prestodb/presto/blob/2ad67dcf000be86ebc5ff7732bbb9994c8e324a8/presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java#L175-L176) gets stuck and runs forever. At least that's where it got stuck in the three separate times I was able to reproduce this locally.

No deadlock, per-se since holding the `DistributedQueryRunner` lock while waiting for the query to finish (which it never does) is by design and locking out other queries that attempt to acquire the exclusive lock (eg: `::testAccessControl`) is just incidental.
